### PR TITLE
Update queries and mutations

### PR DESCRIPTION
**Pull request by newhouse**
## Description of the change

Some of our queries and mutations are a bit outdated. Let's update them.

Oh, and also add a nice framework for testing and specify that we only support Node 12+

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Fixes https://github.com/anvilco/anvil/issues/3960

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development
- [ ] No previous tests unrelated to the changed code fail in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] At least one reviewer has been requested
- [ ] Changes have been reviewed by at least one other engineer
- [ ] The relevant project board has been selected in Projects to auto-link to this pull request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,34 +2,58 @@
   "name": "@anvilco/anvil",
   "version": "2.13.1",
   "description": "Anvil API Client",
-  "main": "src/index.js",
```
Comment (newhouse) [inline package.json:5]: Lots of shuffling around....but some substance.

File: package.json [hunk 2]
```diff
@@ -2,34 +2,58 @@
...
   },
-  "homepage": "https://github.com/anvilco/node-anvil#readme",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/anvilco/node-anvil.git"
+  },
+  "types": "./types/src/index.d.ts",
+  "main": "src/index.js",
+  "engines": {
+    "node": ">=12"
```
Comment (newhouse) [inline package.json:23]: substance.

File: package.json [hunk 3]
```diff
@@ -2,34 +2,58 @@
...
+  },
   "publishConfig": {
     "registry": "https://registry.npmjs.org",
     "@anvilco:registry": "https://registry.npmjs.org"
   },
+  "files": [
```
Comment (newhouse) [inline package.json:29]: substance.

File: src/graphql/mutations/createEtchPacket.js [hunk 4]
```diff
@@ -2,7 +2,14 @@
...
   eid
   name
+  status
+  isTest
+  isFree
+  containsFillData
+  payload
+  numberRemainingSigners
   detailsURL
+  webhookURL
```
Comment (newhouse) [inline src/graphql/mutations/createEtchPacket.js:12]: These are all the things that are possible. Do we want them all?

File: src/graphql/mutations/forgeSubmit.js [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
+  totalSteps
+  continueURL
+  reviewData
+  completionPercentage
+  isExcluded
+  touchedByUser
+  requestMeta
   createdAt
   updatedAt
+  completedAt
```
Comment (newhouse) [inline src/graphql/mutations/forgeSubmit.js:18]: These are all the things that are possible. Do we want them all?

File: src/graphql/queries/etchPacket.js [hunk 6]
```diff
@@ -2,7 +2,18 @@ const defaultResponseQuery = `{
...
+  isFree
+  containsFillData
+  payload
+  numberRemainingSigners
   detailsURL
+  webhookURL
+  createdAt
+  updatedAt
+  archivedAt
+  completedAt
```
Comment (newhouse) [inline src/graphql/queries/etchPacket.js:16]: These are all the things that are possible. Do we want them all?

File: src/graphql/mutations/forgeSubmit.js (hunk 5 above)
Comment (aalmazan) [reply src/graphql/mutations/forgeSubmit.js:18]: Probably fine to keep unless there's a performance hit with always getting everything all the time?

**Review by aalmazan — APPROVED** [review]

File: src/graphql/mutations/createEtchPacket.js (hunk 4 above)
Comment (benogle) [reply src/graphql/mutations/createEtchPacket.js:12]: Keep these new ones:

`status, isTest, numberRemainingSigners, webhookURL`

File: src/graphql/mutations/forgeSubmit.js (hunk 5 above)
Comment (benogle) [reply src/graphql/mutations/forgeSubmit.js:18]: I'd keep 

`status, continueURL, completedAt`

File: src/graphql/queries/etchPacket.js (hunk 6 above)
Comment (benogle) [reply src/graphql/queries/etchPacket.js:16]: I'd keep 

`status, isTest, numberRemainingSigners, webhookURL, *At`

File: src/graphql/mutations/createEtchPacket.js (hunk 4 above)
Comment (benogle) [reply src/graphql/mutations/createEtchPacket.js:12]: Maybe add the date stamps like the query response too?

File: src/graphql/mutations/forgeSubmit.js [hunk 7]
```diff
@@ -28,23 +38,27 @@ module.exports = {
...
+      $enforcePayloadValidOnCreate: Boolean,
       $currentStep: Int,
       $complete: Boolean,
       $isTest: Boolean,
       $timezone: String,
+      $webhookURL: String,
       $groupArrayId: String,
       $groupArrayIndex: Int,
-      $errorType: String,
+      $errorType: String
```
Comment (benogle) [inline src/graphql/mutations/forgeSubmit.js:49]: Remove the errorType, this is for error injection for testing

**Review by benogle — APPROVED** [review]

**Pull request merged**